pkg/tsdb/cloudwatch/metric_find_query.go

```diff
@@ -389,7 +400,9 @@ func (e *cloudWatchExecutor) handleGetMetrics(ctx context.Context, parameters *s
 	} else {
 		var err error
 		dsInfo := e.getDSInfo(region)
-		if namespaceMetrics, err = e.getMetricsForCustomMetrics(region, namespace, dsInfo.Profile); err != nil {
+
+		p := props{region: region, namespace: namespace, profile: dsInfo.Profile}
+		if namespaceMetrics, err = e.getMetricsForCustomMetrics(p); err != nil {
```

Comment:
This change doesn't improve anything, I think the changes in my branch should be applied, where `e.getDSInfo` is called by `e.getMetricsForCustomMetrics` itself. The reason is that with my changes, the burden on the caller is smaller and you get less complicated code.